Framework/PythonInterface/core/inc/MantidPythonInterface/core/PythonStdoutChannel.h

```diff
@@ -27,4 +36,18 @@ class MANTID_PYTHONINTERFACE_CORE_DLL PythonStdoutChannel : public ConsoleChanne
   /// Constructor for PythonStdoutChannel
   PythonStdoutChannel();
 };
+
+/*
+// TODO: this channel should replace PythonStdoutChannel when we adopt pybind11 because of robust GIL management
```

Comment:
It's not really clear if we will adopt `pybind11` - there is a question whether the port is worth it?

When you say robust GIL management, do you mean automatic?

Reply:
pybind11 class [gil_scoped_acquire](https://github.com/pybind/pybind11/blob/master/include/pybind11/gil.h) is the analogous of our class [GlobalInterpreterLock](https://github.com/mantidproject/mantid/blob/master/Framework/PythonInterface/core/inc/MantidPythonInterface/core/GlobalInterpreterLock.h#L23). This pybind11 class seem to address [a few scenarios](https://github.com/pybind/pybind11/blob/master/include/pybind11/gil.h#L26).